src/test/cls_rbd/test_cls_rbd.cc

```diff
@@ -2743,21 +2742,21 @@ TEST_F(TestClsRbd, group_snap_list_max_return) {
   }
 
   std::vector<cls::rbd::GroupSnapshot> snapshots;
-  ASSERT_EQ(0, group_snap_list(&ioctx, group_id, cls::rbd::GroupSnapshot(), 10, &snapshots));
-  ASSERT_EQ(10U, snapshots.size());
```

Comment:
I think this test case is supposed to test `max_return` aspect of the API -- no matter how many snapshots there are, only up to `max_return` should be returned.  With this change you are conflating it with the internal "how many keys are fetched in one go" implementation detail.

I'd leave this test case as is and add a new one targeted specifically at max_read/last_read behavior.

Reply:
What I got to understand here is, to just move the new implementation changes as a new test. Let me know if I understood it the wrong way.

Reply:
Almost ;)  The new test doesn't need to mimic `max_return` test -- a single `group_snap_list` invocation should be sufficient.